tests/test_classifier/test_learning_curve.py

```diff
@@ -0,0 +1,150 @@
+        """
+
+        X = self.occupancy[[
+                "temperature", "relative_humidity", "light", "C02", "humidity"
+            ]]
+
+        y = self.occupancy['occupancy'].astype(int)
+
+        X = X.view((float, len(X.dtype.names)))
+
+        try:
+            visualizer = LearningCurveVisualizer(LinearSVC(random_state=0), train_sizes=np.linspace(.1, 1.0, 5), 
+                cv=ShuffleSplit(n_splits=100, test_size=0.2, random_state=0), 
+                n_jobs=4)
+            visualizer.fit(X, y)
+            visualizer.poof()
+        except Exception as e:
+            self.fail("error during learning curve: {}".format(e))
+
+        self.assert_images_similar(visualizer)
+
+    def test_learning_curve_model_only(self):
+        """
+        Test learning curve with inputting model only. 
+        """
+
+        X = self.occupancy[[
+                "temperature", "relative_humidity", "light", "C02", "humidity"
+            ]]
+
+        y = self.occupancy['occupancy'].astype(int)
+
+        X = X.view((float, len(X.dtype.names)))
+        
+        try:
+            visualizer = LearningCurveVisualizer(LinearSVC())
+            visualizer.fit(X, y)
+            visualizer.poof()
+        except Exception as e:
+            self.fail("error during learning curve: {}".format(e))
+
+    def test_learning_curve_model_cv_only(self):
+        """
+        Test learning curve with inputting model and cv only.
+        """
+
+        X = self.occupancy[[
+                "temperature", "relative_humidity", "light", "C02", "humidity"
+            ]]
+
+        y = self.occupancy['occupancy'].astype(int)
+
+        X = X.view((float, len(X.dtype.names)))
+        
+        try:
+            visualizer = LearningCurveVisualizer(LinearSVC(),
+                cv=ShuffleSplit(n_splits=100, test_size=0.2, random_state=0))
+            visualizer.fit(X, y)
+            visualizer.poof()
+        except Exception as e:
+            self.fail("error during learning curve: {}".format(e))
+
+    def test_learning_curve_model_trainsize_cv_only(self):
+        """
+        Test learning curve with inputting model, training size, and cv only.
+        """
+
+        X = self.occupancy[[
+                "temperature", "relative_humidity", "light", "C02", "humidity"
+            ]]
+
+        y = self.occupancy['occupancy'].astype(int)
+
+        X = X.view((float, len(X.dtype.names)))
+        
+        try:
+            visualizer = LearningCurveVisualizer(LinearSVC(), 
+                train_sizes=np.linspace(.1, 1.0, 5),
+                cv=ShuffleSplit(n_splits=100, test_size=0.2, random_state=0))
+            visualizer.fit(X, y)
+            visualizer.poof()
+        except Exception as e:
+            self.fail("error during learning curve: {}".format(e))
+
+    def test_learning_curve_bad_trainsize(self):
+        """
+        Test learning curve with bad input for training size.
+        """
+
+        X = self.occupancy[[
```

Comment:
I noticed that you are testing this on the entire data set, I recommend that for speeding these tests up you slice and test the first 1,000 rows or something like that.